app/code/Magento/ImportExport/Model/Import/Source/FileParser/UnsupportedPathException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+class UnsupportedPathException extends \RuntimeException
+{
+    private $path;
+
+    public function __construct($path = '', $message = '')
+    {
+        parent::__construct($message ?: sprintf('Path "%s" is not supported', $path));
+
+        $this->path = $path;
+    }
+
+    public function getPath()
```

Comment:
you don't process this exception anywhere currently. It's "processed" just in Unit tests.

Also, don't think that we want to show some Path information in the successor of LocalizedException, which is shown on UI. Because it's insecure to reveal such internal details.
All we will probably do is just log this exception when happened.